File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/sales/sales-order-header-sales-reason/sales-order-header-sales-reason-lookup-model.ts

export interface ISalesOrderHeaderSalesReasonLookupModel {
  salesOrderID: number;
  salesReasonID: number;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class SalesOrderHeaderSalesReasonLookupModel implements ISalesOrderHeaderSalesReasonLookupModel {
  salesOrderID: number;
  salesReasonID: number;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<SalesOrderHeaderSalesReasonLookupModel>) {
    if (!!init) {
      Object.assign<SalesOrderHeaderSalesReasonLookupModel, Partial<SalesOrderHeaderSalesReasonLookupModel>>(this, init);
    }
  }
}

File: src/AdventureWorks.Application/DataEngine/Sales/SalesOrderHeaderSalesReason/Commands/Queries/GetSalesOrderHeaderSalesReasons/SalesOrderHeaderSalesReasonLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.Sales;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;

namespace AdventureWorks.Application.DataEngine.Sales.SalesOrderHeaderSalesReason.Queries.GetSalesOrderHeaderSalesReasons
{
    public partial class SalesOrderHeaderSalesReasonLookupModel : IHaveCustomMapping
    {
        public int SalesOrderID { get; set; }
        public int SalesReasonID { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.Sales.SalesOrderHeaderSalesReason, SalesOrderHeaderSalesReasonLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<SalesOrderHeaderSalesReasonLookupModel, BaseSalesOrderHeaderSalesReason>().IgnoreMissingDestinationMembers();
        }
    }
}
